src/BoundaryConditions/perturbation_advection_open_boundary_matching_scheme.jl

```diff
@@ -79,42 +81,41 @@ const PAOBC = BoundaryCondition{<:Open{<:PerturbationAdvection}}
     Δt = clock.last_stage_Δt
     Δt = ifelse(isinf(Δt), 0, Δt)
 
-    ūⁿ⁺¹ = getbc(bc, l, m, grid, clock, model_fields)
+    ūⁿ⁺¹    = getbc(bc, l, m, grid, clock, model_fields)
     uᵢⁿ     = @inbounds getindex(u, boundary_indices...)
     uᵢ₋₁ⁿ⁺¹ = @inbounds getindex(u, boundary_adjacent_indices...)
     U = max(0, min(1, Δt / ΔX * ūⁿ⁺¹))
 
     pa = bc.classification.matching_scheme
     τ = ifelse(ūⁿ⁺¹ >= 0, pa.outflow_timescale, pa.inflow_timescale)
-    τ̃ = Δt / τ
+    τ̃ = Δt / τ # last stage Δt normalized by the inflow/output timescale
 
     relaxed_uᵢⁿ⁺¹ = (uᵢⁿ + U * uᵢ₋₁ⁿ⁺¹ + ūⁿ⁺¹ * τ̃) / (1 + τ̃ + U)
-    uᵢⁿ⁺¹ = ifelse(τ == 0, ūⁿ⁺¹, relaxed_uᵢⁿ⁺¹)
+    uᵢⁿ⁺¹         = ifelse(τ == 0, ūⁿ⁺¹, relaxed_uᵢⁿ⁺¹)
 
     @inbounds setindex!(u, uᵢⁿ⁺¹, boundary_indices...)
 
     return nothing
 end
 
-@inline function step_left_boundary!(bc::PAOBC, l, m, boundary_indices, boundary_adjacent_indices, boundary_secret_storage_indices,
+@inline function step_left_boundary!(bc::PAOBC, l, m, boundary_indices, boundary_adjacent_indices,
```

Comment:
```suggestion
@inline function step_left_boundary!(bc::PAOBC, l, m, (i1, j1, k1), (i2, j2, k2),
```

this notation change would make this code clearer for me (and likewise everywhere else). I'm ok with labels other than "1" and "2" as well (eg superscript "b", superscript "a" for "boundary" and "adjacent" could work too)

Reply:
Check out what the new notation is for the right boundary. I kept the `boundary_indices` for the function signature (since that clarifies what it is without comments) but used it to define
```julia
    iᵇᵈ, jᵇᵈ, kᵇᵈ = boundary_indices
    iᵃᵈ, jᵃᵈ, kᵃᵈ = boundary_adjacent_indices
```
inside the function. I went with `iᵇᵈ` instead of `iᵇ` because we sometimes use the `ᵇ` superscript for "bottom" in the code, so I wanted to avoid confusion.

I only did it for the left boundary condition atm and will do it for the rest of the file after we converge on a notation.

Reply:
what does "d" mean?

Reply:
Change it to `iᴮ` for boundary indices (that's the same as it is used in https://github.com/CliMA/Oceananigans.jl/blob/main/src/BoundaryConditions/fill_halo_regions_value_gradient.jl) and `iᴬ` for boundary-adjacent indices.

Reply:
great! merge away